test/integration/http_integration.cc

```diff
@@ -1010,7 +1010,7 @@ void HttpIntegrationTest::testUpstreamProtocolError() {
   FakeRawConnectionPtr fake_upstream_connection = fake_upstreams_[0]->waitForRawConnection();
   // TODO(mattklein123): Waiting for exact amount of data is a hack. This needs to
   // be fixed.
-  fake_upstream_connection->waitForData(187);
+  fake_upstream_connection->waitForData(211);
```

Comment:
Can you explain what exactly the old and new value are here, and why it changed?  Or fix the TODO.

Reply:
Sorry meant to drop a comment here. This change is due to the fact that when the connection manager does not use the remote address (attempts to pull from XFF), but XFF is absent or invalid, we use the real connection remote address. Thus, for loopback unit tests, we now set `x-envoy-internal: true` in certain cases which I think is more correct than what we had before.